src/Analytics/Adapter/GoogleAnalytics.php

```diff
@@ -37,6 +37,19 @@ class GoogleAnalytics extends Adapter
      */
     private string $cid;
 
+    /**
+     * Client IP Address
+     * @var string
+     */
+    private string $ip;
```

Comment:
This private properties should also belong to the abstract class in all adapters.